server/src/main/java/io/pravega/schemaregistry/rules/jsoncompatibility/JsonCompatibilityChecker.java

```diff
@@ -0,0 +1,106 @@
+package io.pravega.schemaregistry.rules.jsoncompatibility;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.rules.CompatibilityChecker;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.pravega.schemaregistry.rules.jsoncompatibility.BreakingChangesStore.BreakingChanges;
+
+public class JsonCompatibilityChecker implements CompatibilityChecker {
+    EnumComparator enumComparator;
+    JsonCompatibilityCheckerUtils jsonCompatibilityCheckerUtils;
+    ObjectTypeComparator objectTypeComparator;
+    NumberComparator numberComparator;
+    StringComparator stringComparator;
+    ArrayTypeComparator arrayTypeComparator;
+    public JsonCompatibilityChecker() {
+        this.enumComparator = new EnumComparator();
+        this.jsonCompatibilityCheckerUtils = new JsonCompatibilityCheckerUtils();
+        this.objectTypeComparator = new ObjectTypeComparator();
+        this.numberComparator = new NumberComparator();
+        this.stringComparator = new StringComparator();
+        this.arrayTypeComparator = new ArrayTypeComparator();
+    }
+
+    @Override
+    public boolean canRead(SchemaInfo readUsing, List<SchemaInfo> writtenUsing) {
+        try {
+            return canReadChecker(readUsing, writtenUsing);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    @Override
+    public boolean canBeRead(SchemaInfo writtenUsing, List<SchemaInfo> readUsing) {
+        return false;
```

Comment:
why are you returning false?
shouldnt it be `canBeRead(readUsing, writtenUsing)`?

Reply:
Done. Yes. Had not implemented it then.

Reply:
Will also add tests for this in the IT for the checker.